gossipd/gossipd.c

```diff
@@ -965,15 +962,20 @@ static void update_local_channel(struct daemon *daemon,
 	if (disable)
 		channel_flags |= ROUTING_FLAGS_DISABLED;
 
-	update = towire_channel_update(tmpctx, &dummy_sig,
+	message_flags = 0;
+	if (htlc_maximum_msat > 0)
+		message_flags |= ROUTING_OPT_HTLC_MAX_MSAT;
```

Comment:
I don't *think* this can be zero?  Even if we end up refreshing old pre-this-patch channels (from the store), we'll just set htlc_maximum_msat to the channel capacity (which is what hc->htlc_maximum_msat defaults to.

Reply:
Moreover, if we don't set the flag, we probably shouldn't append a value.  We might assume that any other future appendings would have their own message_flag, but in theory they wouldn't have to...

Reply:
This is true, as written this will never be zero. I will update to remove it. I mostly added the if switch because it serves as an 'explainer' for why we're setting the message flag to that value (ie concretely links the htlc_maximum_msat to the ROUTING_OPT_HTLC_MAX_MSAT flag), but i guess that's what commit histories and inline comments are for!